### PR TITLE
Tezos rebase on master-brd-app

### PR DESCRIPTION
**Pull request by ewerx**

### CODE REVIEW

File: WalletKitCore/src/crypto/BRCryptoConfig.h [hunk 1]
```diff
@@ -140,6 +140,26 @@ DEFINE_ADDRESS_SCHEMES  ("hedera-testnet", CRYPTO_ADDRESS_SCHEME_GEN_DEFAULT)
...
+DEFINE_NETWORK_FEE_ESTIMATE ("tezos-mainnet", "500000", "1m", 1 * 60 * 1000)
+DEFINE_CURRENCY ("tezos-mainnet",     "tezos-mainnet:__native__",   NETWORK_NAME,  CRYPTO_NETWORK_CURRENCY_XTZ,  "native",   NULL,   true)
+    DEFINE_UNIT ("tezos-mainnet:__native__",  "mutez",       "mtz",   0,  "mutez")
+    DEFINE_UNIT ("tezos-mainnet:__native__",  NETWORK_NAME,  "xtz",   6,  "XTZ")
+DEFINE_ADDRESS_SCHEMES  ("tezos-mainnet", CRYPTO_ADDRESS_SCHEME_GEN_DEFAULT)
+DEFINE_MODES            ("tezos-mainnet", CRYPTO_SYNC_MODE_API_ONLY)
+
+DEFINE_NETWORK (xtzTestnet,  "tezos-testnet", NETWORK_NAME, "testnet", false, 50000, 1)
+DEFINE_NETWORK_FEE_ESTIMATE ("tezos-testnet", "500000", "1m", 1 * 60 * 1000)
+DEFINE_CURRENCY ("tezos-testnet",     "tezos-testnet:__native__",   NETWORK_NAME,  CRYPTO_NETWORK_CURRENCY_XTZ,  "native",   NULL,   true)
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoConfig.h:156]: No `HAS_XTZ_TESTNET` here yet.  Okay.

File: WalletKitCore/src/crypto/BRCryptoHash.c [hunk 2]
```diff
@@ -99,7 +105,10 @@ cryptoHashString (BRCryptoHash hash) {
             return ethHashAsString (hash->u.eth);
         }
         case BLOCK_CHAIN_TYPE_GEN: {
-            return _cryptoHashAddPrefix (genericHashAsString(hash->u.gen), 1);
+            char *string = genericHashAsString (hash->u.gen);
+            return (GENERIC_HASH_ENCODING_HEX == hash->u.gen.encoding)
+            ? _cryptoHashAddPrefix (string, 1)
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoHash.c:110]: `string` is not leaked.  Okay.

File: WalletKitCore/src/crypto/BRCryptoWalletManager.c [hunk 3]
```diff
@@ -943,15 +944,18 @@ cryptoWalletManagerSign (BRCryptoWalletManager cwm,
...
             break;
 
-        case BLOCK_CHAIN_TYPE_GEN:
+        case BLOCK_CHAIN_TYPE_GEN: {
+            BRGenericHash lastBlockHash = cryptoHashAsGEN (cryptoNetworkGetVerifiedBlockHash (cwm->network));
             success = AS_CRYPTO_BOOLEAN (genManagerSignTransfer (cwm->u.gen,
                                                                  cryptoWalletAsGEN (wallet),
+                                                                 lastBlockHash,
                                                                  cryptoTransferAsGEN (transfer),
                                                                  seed));
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoWalletManager.c:953]: `cryptoNetworkGetVerifiedBlockHash` does `cryptoHashTake` - it is never given?

File: WalletKitCore/src/crypto/BRCryptoWalletManager.c [hunk 4]
```diff
@@ -995,9 +999,10 @@ cryptoWalletManagerSubmit (BRCryptoWalletManager cwm,
         case BLOCK_CHAIN_TYPE_GEN: {
             BRGenericWallet genWallet = cryptoWalletAsGEN (wallet);
             BRGenericTransfer genTransfer = cryptoTransferAsGEN (transfer);
+            BRGenericHash lastBlockHash = cryptoHashAsGEN (cryptoNetworkGetVerifiedBlockHash (cwm->network));
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoWalletManager.c:1002]: Same as above, the `VerifiedBlockHash` is never given?

File: WalletKitCore/src/crypto/BRCryptoWalletManager.c [hunk 5]
```diff
@@ -1295,29 +1302,54 @@ cryptoWalletManagerEstimateFeeBasis (BRCryptoWalletManager cwm,
 
             UInt256 genValue = cryptoAmountGetValue(amount);
             UInt256 genPricePerCostFactor = uint256Create (cryptoNetworkFeeAsGEN(fee));
+            
+            if (genManagerSupportsLocalFeeEstimation(cwm->u.gen)) {
+                BRGenericFeeBasis genFeeBasis = genWalletEstimateTransferFee (genWallet,
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoWalletManager.c:1307]: Ugh, not 'refactor'.  But, okay.

File: WalletKitCore/src/generic/BRGeneric.c [hunk 6]
```diff
@@ -118,9 +128,11 @@ genAccountGetSerialization (BRGenericAccount account, size_t *bytesCount) {
 
 extern void
 genAccountSignTransferWithSeed (BRGenericAccount account,
+                                BRGenericWallet wallet,
+                                BRGenericHash lastBlockHash,
```
Comment (EBGToo) [inline WalletKitCore/src/generic/BRGeneric.c:132]: In 'refactor/develop' feels like we will need a 'signing context' concept that is currency specific.

File: WalletKitCore/src/crypto/BRCryptoWalletManager.c (hunk 4 above)
Comment (ewerx) [reply WalletKitCore/src/crypto/BRCryptoWalletManager.c:1002]: good catch, fixed

**Review by EBGToo — APPROVED** [review]

**Pull request merged**